packages/library-base/src/commonMain/kotlin/io/realm/internal/BaseRealmImpl.kt

```diff
@@ -66,25 +68,29 @@ abstract class BaseRealmImpl internal constructor(
         // Use same reference through out all operations to avoid locking
         val realmReference = this.realmReference
         realmReference.checkClosed()
-        return RealmResultsImpl.fromQuery(
-            realmReference,
-            RealmInterop.realm_query_parse(
-                realmReference.dbPointer,
-                clazz.simpleName!!,
-                "TRUEPREDICATE"
-            ),
-            clazz,
-            configuration.mediator
+        val findAllQuery = RealmInterop.realm_query_parse(
```

Comment:
Instead of doing all of this, I would call into the publicAPI of `query`. That way we only have one internal codepath.